qemu/tests/readonly_floppy.py

```diff
@@ -24,8 +26,16 @@ def run(test, params, env):
     :param env: Dictionary with test environment.
     """
 
-    error.context("Boot up guest with floppies", logging.info)
+    vt_data_dir = data_dir.get_data_dir()
+    pre_command_noncritical = params.get("pre_command_noncritical") == "yes"
```

Comment:
I think update params["pre_cmd"] -> set params["start_vm"]=yes ->env_process.preprocess() will be better.

Reply:
ok , will update, thx.